Bottom: 9cc84e0a1cbe424b99b72c9d5a2df1c5e82ceed3
Top:    1172dccc3a9307a18d1073788ccc20d3648ee8a9
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-05-22 20:43:09 -0400

Refresh of temp

---

diff --git a/sound/soc/fsl/mpc5200_psc_ac97.c b/sound/soc/fsl/mpc5200_psc_ac97.c
index 39af12d..7253303 100644
--- a/sound/soc/fsl/mpc5200_psc_ac97.c
+++ b/sound/soc/fsl/mpc5200_psc_ac97.c
@@ -362,15 +362,19 @@ reset:
 	timeout = 0;
 	while ((in_8(&regs->ipcr_acr.ipcr) & 0x80) != 0) {
 		udelay(1);
-		if (timeout++ > 1000)
+		if (timeout++ > 1000) {
+			psc_ac97_warm_reset(&ac97);
 			goto reset;
+		}
 	}
 	/* then wait for the transition to high */
 	timeout = 0;
 	while ((in_8(&regs->ipcr_acr.ipcr) & 0x80) == 0) {
 		udelay(1);
-		if (timeout++ > 1000)
+		if (timeout++ > 1000) {
+			psc_ac97_warm_reset(&ac97);
 			goto reset;
+		}
 	}
 
 	out_8(&regs->command, MPC52xx_PSC_TX_ENABLE | MPC52xx_PSC_RX_ENABLE);
